### PR TITLE
Refactor AuthKeyPair.raw_create_key_pair

### PR DESCRIPTION
**Pull request by agrare**
Merge together with: https://github.com/ManageIQ/manageiq/pull/21759

### CODE REVIEW

File: app/models/manageiq/providers/openstack/cloud_manager/auth_key_pair.rb [hunk 1]
```diff
@@ -7,7 +7,9 @@ class ManageIQ::Providers::Openstack::CloudManager::AuthKeyPair < ManageIQ::Prov
   def self.raw_create_key_pair(ext_management_system, create_options)
     connection_options = {:service => 'Compute'}
     ext_management_system.with_provider_connection(connection_options) do |service|
-      service.key_pairs.create(create_options)
+      kp = service.key_pairs.create(create_options)
+
+      {:name => kp.name, :fingerprint => kp.fingerprint, :auth_key => kp.private_key}
```
Comment (jaywcarman) [inline app/models/manageiq/providers/openstack/cloud_manager/auth_key_pair.rb:12]: I think this is only used to store public keys right?
```suggestion
      {:name => kp.name, :fingerprint => kp.fingerprint, :auth_key => kp.public_key}
```

Comment (agrare) [reply app/models/manageiq/providers/openstack/cloud_manager/auth_key_pair.rb:12]: `:auth_key` is used to store the private key, see https://github.com/ManageIQ/manageiq/pull/21759/files#diff-b80f64b28488206bf8d41b9b53fb77a5490267a1a91b5d1e4f0897af88951554L46

Comment (jaywcarman) [reply app/models/manageiq/providers/openstack/cloud_manager/auth_key_pair.rb:12]: Thanks for the explanation! You're right - I was looking at the `:public_key` column. Makes sense now, though :)

**Comment by agrare** [comment]
Test failures are due to codependency on the core PR, cross-repo tests are green

**Review by Fryguy — APPROVED** [review]

**Comment by Fryguy** [comment]
Backported to `najdorf` in commit f66db09632fbc7df77a096f7862d05e6cc541387.

```text
commit f66db09632fbc7df77a096f7862d05e6cc541387
Author: Jason Frey <fryguy9@gmail.com>
Date:   Tue Mar 8 14:45:55 2022 -0500

    Merge pull request #784 from agrare/refactor_raw_create_key_pair
    
    Refactor AuthKeyPair.raw_create_key_pair
    
    (cherry picked from commit b72699ade55e08d4a2e5ecd87f621dbe0a8be1fc)
```

**Pull request merged**